controllers/osccluster_securitygroup_controller.go

```diff
@@ -298,7 +298,7 @@ func reconcileSecurityGroup(ctx context.Context, clusterScope *scope.ClusterScop
 
 		if !Contains(securityGroupIds, securityGroupId) && tag == nil {
 
-			if extraSecurityGroupRule {
+			if extraSecurityGroupRule && len(clusterScope.GetSecurityGroupsRef().ResourceMap) == len(clusterScope.GetSecurityGroups()) {
```

Comment:
I think it's better to break it down into smaller, descriptive boolean len(securityGroupsRef)==len(securityGroups)
Also what if securityGroupsRef == nil || securityGroups == nil ? 
Maybe it's better to test it to avoid runtime errors

Reply:
Because of the above code, securityGroupsRef and securityGroups will never be null.

Reply:
fixed

Reply:
fixed